packages/langium/src/workspace/workspace-manager.ts

```diff
@@ -130,10 +144,15 @@ export class DefaultWorkspaceManager implements WorkspaceManager {
         // we can still assume that all library documents and file documents are loaded by the time we start building documents.
         // The mutex prevents anything from performing a workspace build until we check the cancellation token
         await this.loadAdditionalDocuments(folders, collector);
+        const uris: URI[] = [];
         await Promise.all(
-            folders.map(wf => [wf, this.getRootFolder(wf)] as [WorkspaceFolder, URI])
-                .map(async entry => this.traverseFolder(...entry, fileExtensions, collector))
+            folders.map(wf => this.getRootFolder(wf))
+                .map(async entry => this.traverseFolder(entry, uris))
         );
+        await Promise.all(uris.map(async uri => {
+            const document = await this.langiumDocuments.getOrCreateDocument(uri);
+            collector(document);
+        }));
```

Comment:
This can be simplified now: instead of using a collector function, just map the uris to the created documents.

Reply:
Ok let's keep it as it is.